homeassistant/helpers/config_validation.py

```diff
@@ -450,10 +460,11 @@ def validator(value):
 
 TIME_CONDITION_SCHEMA = vol.All(vol.Schema({
     vol.Required(CONF_CONDITION): 'time',
+    'day': date,
```

Comment:
You cannot update the time condition schema without also updating the time condition code (and adding a test for that)